Fix issue of trying to create new PDFs in preexisting directory.

dustin-lennon opened the pull request:
PDFs would fail to be created due to save directory already existing and current code cannot do a make directory with a folder of the same name. Pull request checks if directory exists and if it does not creates the directory and does the save to the directory. If the directory does exist, saves the PDF to the directory.

Commit:
Fix issue of trying to create new PDFs in preexisting directory.

Commit:
Revert changes pushed to master for a clean branch

Commit:
Pass addtional parameters to pdftk application

Update index.js, pdfGenerator.js, and README.md file to use and explain how to pass additional parameters to pdftk command

dustin-lennon committed:
Merge pull request #1 from demonicpagan/pdftk-additional-options

Pass addtional parameters to pdftk application

dustin-lennon committed:
Remove console.logs used for debugging purposes

dustin-lennon committed:
Merge pull request #2 from demonicpagan/pdftk-additional-options

Remove console.logs used for debugging purposes

Commit:
Original files for base master

Commit:
Check if folder exists before attempting to save file

Commit:
Revert README to original

dustin-lennon committed:
Merge pull request #3 from demonicpagan/save-file-fix

Save file fix

Commit:
Push original index.js to master

Commit:
Merge changes of pdftk-additional-options into master

txssseal commented:
@deejayres @sdiachenko @B-3PO Some additonal pr's from people using this library.  May help, please see if this will cause any breaking changes.

deejayres commented:
@absalas

txssseal commented on lib/pdfGenerator.js:
```diff
@@ -22,25 +22,39 @@ module.exports = function generatePDF(options, templates, fileName, pdftkPath) {
+
+  console.log('Files: ', files);
+  console.log('PDFtk Options: ', pdftkOptions);
+  console.log('Function test: ', convertObjString(pdftkOptions));
```
can we get rid of the console logs?

dustin-lennon replied:
done

txssseal commented on lib/saveFile.js:
```diff
@@ -1,21 +1,35 @@
-        console.error(err)
-        return;
-      }
+    // Check if directory being written to already exists
```
:clap:

txssseal commented on lib/pdfGenerator.js:
```diff
@@ -5,14 +5,14 @@ var exec = require('child_process').exec;
 var pexec = Promise.promisify(exec);
 
-module.exports = function generatePDF(options, templates, fileName, pdftkPath) {
+module.exports = function generatePDF(options, templates, fileName, pdftkPath, pdftkOptions) {
```
@absalas in some of your changes you will need to take into account his new changes

txssseal replied:
specifically passing in the whole options object

dustin-lennon committed:
Remove console.logs

dustin-lennon committed:
Missed a console.log to remove

deejayres commented:
Heyo, we're in the middle of the process of deploying features that depend on this, and are trying to minimize the moving parts until we can flex on this in prod.  Honestly had no idea this was an open repo. We can keep the PR open, but it might be a while before we can approve.

Commit:
fix: .snyk & package.json to reduce vulnerabilities


The following vulnerabilities are fixed with a Snyk patch:
- https://snyk.io/vuln/SNYK-JS-LODASH-450202

dustin-lennon committed:
Merge pull request #4 from dustin-lennon/snyk-fix-58a5bf542bfa7d2afeabf4579e17f5b8

[Snyk] Fix for 1 vulnerable dependencies

Commit:
fix: .snyk & package.json to reduce vulnerabilities


The following vulnerabilities are fixed with a Snyk patch:
- https://snyk.io/vuln/SNYK-JS-LODASH-450202

dustin-lennon committed:
Merge pull request #5 from dustin-lennon/snyk-fix-0960c072e3ab3fa4a2be2ec803451aab

[Snyk] Fix for 1 vulnerable dependencies

aybhalala commented:
Any ETA when will this be merged?

snyk-bot committed:
fix: package.json to reduce vulnerabilities

The following vulnerabilities are fixed with an upgrade:
- https://snyk.io/vuln/SNYK-JS-UGLIFYJS-1727251

snyk-bot committed:
fix: package.json to reduce vulnerabilities

The following vulnerabilities are fixed with an upgrade:
- https://snyk.io/vuln/SNYK-JS-HTMLPDF-467248

snyk-bot committed:
fix: package.json to reduce vulnerabilities

The following vulnerabilities are fixed with an upgrade:
- https://snyk.io/vuln/SNYK-JS-EJS-6689533

snyk-bot committed:
fix: package.json to reduce vulnerabilities

The following vulnerabilities are fixed with an upgrade:
- https://snyk.io/vuln/SNYK-JS-PUGCODEGEN-7086056

dustin-lennon committed:
Merge pull request #16 from dustin-lennon/snyk-fix-7dfa140111b205cf33933ca6dd945830

[Snyk] Security upgrade pug from 2.0.4 to 3.0.0

dustin-lennon committed:
Merge pull request #15 from dustin-lennon/snyk-fix-81db38ff88842757f5f9a1e42bae1eea

[Snyk] Security upgrade ejs from 2.7.4 to 3.1.10

dustin-lennon committed:
Merge pull request #14 from dustin-lennon/snyk-fix-2e967ec48549ac3078eab14c0080e9fe

[Snyk] Security upgrade html-pdf from 2.2.0 to 3.0.1

dustin-lennon committed:
Merge pull request #11 from dustin-lennon/snyk-fix-9345d3cfe650172c09584e1f5e733198

[Snyk] Security upgrade pug from 2.0.4 to 3.0.0
